pyanaconda/ui/webui/__init__.py

```diff
@@ -141,14 +141,38 @@ def _run_webui(self):
         # FIXME: We probably want to move this to early execution similar to what we have on live
         profile_name = FIREFOX_THEME_DEFAULT
 
-        proc = startProgram(["/usr/libexec/anaconda/webui-desktop",
-                             "-t", profile_name, "-r", str(int(self.remote)),
-                             "/cockpit/@localhost/anaconda-webui/index.html"],
-                            reset_lang=False)
-        log.debug("cockpit web view has been started")
-        with open(self._viewer_pid_file, "w") as f:
-            f.write(repr(proc.pid))
-        proc.wait()
+        try:
+            proc = startProgram(
+                ["/usr/libexec/anaconda/webui-desktop",
+                 "-t", profile_name, "-r", str(int(self.remote)),
+                 "/cockpit/@localhost/anaconda-webui/index.html"],
+                reset_lang=False
+            )
+
+            log.debug("cockpit web view has been started")
+            with open(self._viewer_pid_file, "w") as f:
+                f.write(repr(proc.pid))
+
+            (output_string, err_string) = proc.communicate()
```

Comment:
The issue I'm seeing (if I read it correctly) is that you are blocking completely this thread. If this thread is the main thread (which I guess it is) it will also block event loop so no events are processed. That could have a problematic outcomes like not being able to react on exceptions and react on DBus events.

Please verify if my assumptions are correct or wrong.

Reply:
There shouldn't be any difference from the original code, there was a wait in the same place, which blocked the kernel until the webui was finished. 
The BOSS who takes care of this should have been running in a parallel process long ago and this should not affect him in any way.

Reply:
I somehow remember that I said even back then that this is not correct way to handle it. But if it was already part of the original code I guess it should be fine here....